apigw/src/internal/app.ts

```diff
@@ -42,148 +37,113 @@ import { cacheControl } from '../shared/middleware/cache-control.js'
 import { createSamlConfig } from '../shared/saml/index.js'
 import redisCacheProvider from '../shared/saml/passport-saml-cache-redis.js'
 import { createDevAdRouter } from './dev-ad-auth.js'
-import { assertRedisConnection, RedisClient } from '../shared/redis-client.js'
+import { RedisClient } from '../shared/redis-client.js'
 
-export default function internalGwApp(
+export function internalGwRouter(
   config: Config,
   redisClient: RedisClient
-) {
-  const app = express()
-  trustReverseProxy(app)
-  app.set('etag', false)
+): Router {
+  const router = Router()
 
-  app.use(
+  router.use(session('employee', redisClient))
+  router.use(touchSessionMaxAge)
+  router.use(passport.session())
+  router.use(cookieParser(cookieSecret))
+  router.use(refreshLogoutToken())
+
+  router.use(
     cacheControl((req) =>
-      req.path.startsWith('/api/internal/child-images/')
-        ? 'allow-cache'
-        : 'forbid-cache'
+      req.path.startsWith('/child-images/') ? 'allow-cache' : 'forbid-cache'
     )
   )
 
-  app.use(
-    helmet({
-      // Content-Security-Policy is set by the nginx proxy
-      contentSecurityPolicy: false
-    })
-  )
-  app.get('/health', (_, res) => {
-    assertRedisConnection(redisClient)
-      .then(() => {
-        res.status(200).json({ status: 'UP' })
-      })
-      .catch(() => {
-        res.status(503).json({ status: 'DOWN' })
-      })
-  })
-  app.use(tracing)
-  app.use(session('employee', redisClient))
-  app.use(touchSessionMaxAge)
-  app.use(cookieParser(cookieSecret))
-  app.use(passport.initialize())
-  app.use(passport.session())
-  passport.serializeUser<Express.User>((user, done) => done(null, user))
-  passport.deserializeUser<Express.User>((user, done) => done(null, user))
-  app.use(refreshLogoutToken())
-  setupLoggingMiddleware(app)
-
-  app.use('/api/csp', csp)
-
-  function internalApiRouter() {
-    const router = Router()
-    router.all('/system/*', (_, res) => res.sendStatus(404))
+  router.all('/system/*', (_, res) => res.sendStatus(404))
 
-    const integrationUsers = {
-      ...(titaniaConfig && {
-        [titaniaConfig.username]: titaniaConfig.password
-      }),
-      ...(espooBiPocPassword && { 'espoo-bi-poc': espooBiPocPassword })
-    }
-    router.use('/integration', expressBasicAuth({ users: integrationUsers }))
-    router.all('/integration/*', createProxy())
-
-    router.all('/auth/*', (req: express.Request, res, next) => {
-      if (req.session?.idpProvider === 'evaka') {
-        req.url = req.url.replace('saml', 'evaka')
-      }
-      next()
-    })
+  const integrationUsers = {
+    ...(titaniaConfig && {
+      [titaniaConfig.username]: titaniaConfig.password
+    }),
+    ...(espooBiPocPassword && { 'espoo-bi-poc': espooBiPocPassword })
```

Comment:
Could the words espoo and poc be replaced with something more production like, for example just BiPassword?